FabLabKasse/faucardPayment/generateLog.py

```diff
@@ -144,7 +144,7 @@ def verify_sum(curKb, start, end, magposSum):
                 lastdate = timestamp
 
             # Determine corresponding rechnung
-            curKb.execute("SELECT rechnung, datum FROM buchung WHERE konto = 'FAUKarte' AND (betrag = ? OR betrag = ? OR betrag = ? OR betrag = ? OR betrag = ? OR betrag = ?) AND datum BETWEEN ? AND ? ",(unicode(amount),"{:.0f}".format(amount),"{:.1f}".format(amount), "{:.2f}".format(amount), "{:.3f}".format(amount), "{:.4f}".format(amount), timestamp, timestamp + timedelta(seconds=20)))
+            curKb.execute("SELECT rechnung, datum FROM buchung WHERE konto = 'FAUKarte' AND (abs(betrag - ?) < 1e-4) AND datum BETWEEN ? AND ? ",(unicode(amount), timestamp, timestamp + timedelta(seconds=20)))
```

Comment:
It is amusing that this SQL statement really works if `betrag` is  column type `TEXT`. Let's consider it a feature of SQLite :-D